main.js

```diff
@@ -735,28 +751,48 @@ function main() {
         lineBetween.getCenter(origin);
         bisector.crossVectors(pointDirection, new THREE.Vector3(0, 0, 1));
 
-        let intersections = lineAllCollisions(origin, bisector);
-        let mindex = 0;
-        let maxdex = 0;
-        for (let i = 0; i < intersections.length; i++) {
-            if (intersections[i][0] < intersections[mindex][0]) {
-                mindex = i;
-            } else if (intersections[i][1] < intersections[mindex][1]) {
-                mindex = i;
-            }
-            if (intersections[i][0] > intersections[maxdex][0]) {
-                maxdex = i;
-            } else if (intersections[i][1] > intersections[maxdex][1]) {
-                maxdex = i;
-            }
-        }
+        let newVerts = findLineLimits(origin, bisector);
 
-        createNewEdge(new THREE.Vector3(intersections[mindex][0], intersections[mindex][1], 0),
-            new THREE.Vector3(intersections[maxdex][0], intersections[maxdex][1], 0));
+        createNewEdge(newVerts[0], newVerts[1]);
     }
 
+    // find bisector for the two lines, then intersection point, take this as the crease
     function foldAxiom3() {
-        console.warn("Axiom 3 not implemented");
+        let line1 = mathLines[selectedLines[0]];
+        let line2 = mathLines[selectedLines[1]];
+        // find the direction of the bisecting line
+        let bisector = new THREE.Vector3();
+        let dir1 = new THREE.Vector3();
+        line1.delta(dir1);
+        let dir2 = new THREE.Vector3();
+        line2.delta(dir2);
+
+        // we need the two vectors to be pointing in the same direction
+        if (dir1.angleTo(dir2) > (Math.PI / 2)) {
```

Comment:
Another way of doing this, BTW, is to compute the dot product of dir1 and dir2 and compare it against zero. (In particular, I think the current version will fail if `dir1.angleTo(dir2)` is less than -PI/2 (I'm presuming the angle range is between -PI and PI) )